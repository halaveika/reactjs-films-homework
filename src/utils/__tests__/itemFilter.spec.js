import itemfilter from '../itemFilter';

const item = [{
  id: 2412412,
  title: 'Fight club',
  genres: ['drama', 'sport', 'comedy'],
  vote_average: 8,
  poster: 'image_path',
  overview: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry',
},
{
  id: 41412,
  title: 'Sopranos',
  genres: ['drama', 'Criminal', 'comedy'],
  vote_average: 10,
  poster: 'image_path',
  overview: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry',
}];

describe('itemfilter', () => {
  test('itemfilter should filter array', async () => {
    const result = itemfilter(item, 'Criminal');
    expect(result).toEqual([item[1]]);
    expect(result).toMatchSnapshot();
  });

  test('itemfilter should not filter array if there are not pass genre', async () => {
    const result = itemfilter(item, '');
    expect(result).toEqual(item);
    expect(result).toMatchSnapshot();
  });
});
