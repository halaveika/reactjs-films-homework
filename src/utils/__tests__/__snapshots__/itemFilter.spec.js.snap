// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`itemfilter itemfilter should filter array 1`] = `
Array [
  Object {
    "genres": Array [
      "drama",
      "Criminal",
      "comedy",
    ],
    "id": 41412,
    "overview": "Lorem Ipsum is simply dummy text of the printing and typesetting industry",
    "poster": "image_path",
    "title": "Sopranos",
    "vote_average": 10,
  },
]
`;

exports[`itemfilter itemfilter should not filter array if there are not pass genre 1`] = `
Array [
  Object {
    "genres": Array [
      "drama",
      "sport",
      "comedy",
    ],
    "id": 2412412,
    "overview": "Lorem Ipsum is simply dummy text of the printing and typesetting industry",
    "poster": "image_path",
    "title": "Fight club",
    "vote_average": 8,
  },
  Object {
    "genres": Array [
      "drama",
      "Criminal",
      "comedy",
    ],
    "id": 41412,
    "overview": "Lorem Ipsum is simply dummy text of the printing and typesetting industry",
    "poster": "image_path",
    "title": "Sopranos",
    "vote_average": 10,
  },
]
`;
